modules/nextflow/src/main/groovy/nextflow/cli/CmdRun.groovy

```diff
@@ -405,7 +405,7 @@ class CmdRun extends CmdBase implements HubOptions {
     }
 
     private static boolean ignoreHistory() {
-        System.getenv('NXF_IGNORE_RESUME_HISTORY')=='true'
+        System.getenv().getOrDefault('NXF_IGNORE_RESUME_HISTORY',"false").toBoolean()=='true'
```

Comment:
I think this is wrong because compare `Boolean` with a string therefore it will be always `false`. 

I think it would be better to only change  `NXF_OFFLINE` using the same "pattern" `System.getenv('NXF_OFFLINE')=='true' which is a bit more safe, because  the `toBoolean` could thrown an exception for a non `true|false` string

Reply:
toBoolean doesnt throw any exception, only returns `true` if the string is "true", "1"  or "y" 

https://docs.groovy-lang.org/latest/html/groovy-jdk/java/lang/String.html#toBoolean()

Reply:
I dont know this, and actually is nice, but my concern is to have those kinds of subtle changes across versions and attributes how boolean is interpreted.  

Maybe we should look for variables and change all to use `toBoolean` but I think it' outside the scope of this patch